### PR TITLE
Add min/max-imize button to gr.Image and gr.Gallery

### PR DESCRIPTION
**Pull request by hannahblair**
## Description

Allows users to maximise and minimise an image of an output component to full screen.

It could also be good to include zoom in and out with panning outside of this PR to improve the UX of zoom functionality - I did initially include zoom in/out but its bad UX without the panning. 

~❓ question! should this be a param? if so, what should be the default? ❓~
~❓ another question! any other components we should apply this to? annotated image?❓~

Closes: #6939
Closes: #4908

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by abidlabs** [comment]
This is awesome @hannahblair! UX looks good to me

> ❓ question! should this be a param? if so, what should be the default? ❓

Yes, I think so. Similar to e.g. show_share_button, you could have a `show_fullscreen_button` (which is by default `True` to help with discoverability)

> ❓ another question! any other components we should apply this to? annotated image?❓

Yes, I think this should be part of both `gr.AnnotatedImage` and `gr.ImageEditor` when `interactive=False`

File: js/annotatedimage/Index.svelte [hunk 1]
```diff
@@ -38,6 +39,24 @@
...
+	let image_container: HTMLElement;
+
+	onMount(() => {
+		document.addEventListener("fullscreenchange", () => {
+			is_full_screen = !!document.fullscreenElement;
+		});
+	});
+
+	const toggle_full_screen = async (): Promise<void> => {
+		if (!document.fullscreenElement) {
```
Comment (freddyaboulton) [inline js/annotatedimage/Index.svelte:54]: nit: use `is_full_screen` here for consistency

**Review by freddyaboulton — COMMENTED** [review]
Looks good @hannahblair - tested demos locally and in the deployed space and the functionality works as expected.

I think we should use the same minimize icon in the chatbot full screen mode for consistency. Right now an `X` is used.

image

Actually, the chatbot already has some custom behavior for having images/gallery take up the entire chatbot area. I think we should remove and use this behavior in this PR. Thoughts @dawoodkhan82 ?

Also, should we also add this parameter for video?

**Comment by freddyaboulton** [comment]
> This would actually be a nice idea! I think its worth doing this in a separate PR, alongside the updated minimise icon

Agreed!

**Review by abidlabs — APPROVED** [review]
lgtm great implementation @hannahblair!

**Comment by pngwn** [comment]
rude

**Comment by FurkanGozukara** [comment]
finally full screen arrived nice

do we also have no full screen image slider?

**Comment by abidlabs** [comment]
Full screen would be really useful for the ImageSlider agreed, I’ll make a note on the issue

**Pull request merged**